src/AeroShield.cpp

```diff
@@ -0,0 +1,172 @@
+/*
+  API for the AeroShield hardware.
+  
+  The file is a part of the application programming interface for
+  the AeroShield didactic tool for control engineering and 
+  mechatronics education.
+  
+  This code is part of the AutomationShield hardware and software
+  ecosystem. Visit http://www.automationshield.com for more
+  details. This code is licensed under a Creative Commons
+  Attribution-NonCommercial 4.0 International License.
+
+  Created by .... 
+  Last update: 02.03.2022.
+*/
+
+#include "AeroShield.h"         // Include header file
+
+// Initializes hardware pins
+float AeroClass::begin(bool isDetected){                             // Board initialisation
+    pinMode(AERO_UPIN,OUTPUT);  		                                  // Actuator pin
+
+  #ifdef ARDUINO_ARCH_AVR                                             // For AVR architecture boards
+    Wire.begin();                                                     // Use Wire object
+  #elif ARDUINO_ARCH_SAM                                              // For SAM architecture boards
+    Wire1.begin();                                                    // Use Wire1 object
+  #elif ARDUINO_ARCH_SAMD                                             // For SAMD architecture boards
+    Wire.begin();                                                     // Use Wire object
+  #endif
+        if(isDetected == 0 ){                                           // If magnet not detected go on
+    while(1){                                                           // Go forever until magnet detected 
+        if(isDetected == 1 ){                                           // If magnet detected
+            AutomationShield.serialPrint("Magnet detected \n");         // Print information then break
+            break;
+        }
+        else{                                                           // If magnet not detected 
+            AutomationShield.serialPrint("Can not detect magnet \n");   // Print information then go back to check while statement
+      }
+    }
+  }       
+} 
+
+float AeroClass::convertRawAngleToDegrees(word newAngle) {             // Function for converting raw angle(0-4096) to degrees(0-360°) 
+  float retVal = newAngle * 0.087;                                      // 360°/4096=0.087° times the raw value
+  ang = retVal;                               
+  return ang;                                                           // Return angle value in degrees 
+}
+
+float AeroClass::calibration(word RawAngle) {                          // Calibration 
+  AutomationShield.serialPrint("Calibration running...\n");             // Print info 
+  startangle=0;                                                         // Zero out Variable(precaution)
```

Comment:
Variables must be written in camelCase (https://en.wikipedia.org/wiki/Camel_case)